be/src/services/vocabulary.service.ts

```diff
@@ -26,6 +27,53 @@ export class VocabularyService {
         return this.collection.findOne({_id: new ObjectId(id)}).then(x => <Vocabulary>x);
     }
 
+    public updateVocab(id: string, ifUnmodifiedSince: Date, newVocab: Vocabulary): Promise<Vocabulary> {
+        // @ts-ignore
+        return this.getVocabular(id)
+          .then((dbo: Vocabulary) => {
+              if(!!dbo) {
+                  if(dbo.lastModified.getTime() <= ifUnmodifiedSince.getTime()) {
+                      const filter: Filter<Vocabulary> = {
+                          _id: new ObjectId(id),
+                          lastModified: {
+                              // eslint-disable-rows-line @typescript-eslint/no-unsafe-argument
+                              $lte: ifUnmodifiedSince
+                          }
+                      };
+
+                      const update: UpdateFilter<Vocabulary> = {
+                          $set: {
+                              _id: new ObjectId(id),
+                              label: newVocab.label,
+                              description: newVocab.description,
+                              lastModified: new Date()
+                          }
+                      };
+
+                      // @ts-ignore
+                      return this.collection.updateOne(filter, update, {upsert: false})
```

Comment:
Check out the operation `findAndModify` which allows you combining the update and retrieval request in one query.

https://mongodb.github.io/node-mongodb-native/4.2/classes/Collection.html#findOneAndUpdate

Reply:
> Check out the operation `findAndModify` which allows you combining the update and retrieval request in one query.
> 
> https://mongodb.github.io/node-mongodb-native/4.2/classes/Collection.html#findOneAndUpdate

@ja-fra With `$eq: ifUnmodifiedSince`set `findAndModify` will not find the entry if it has changed in the meantime. So I the capability of erroring with `412, 'Precondition Failed'` will be lost and hidden in `404, 'Not found'`. Is that ok?

Reply:
Nope. In the error-case you would need to check why it failed in a separate query: `404` or `412`